modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -569,7 +574,42 @@ private boolean shouldUsePrecalculatedContext(CoreEvent event) {
   }
 
   protected void executeOperation(ExecutionContextAdapter<T> operationContext, ExecutorCallback callback) {
-    executionMediator.execute(componentExecutor, operationContext, callback);
+    final MutableConfigurationStats stats = getMutableConfigurationStats(operationContext);
```

Comment:
feels like all of this should be happening inside the mediator. Actually, if you look at the mediators implementation, it will obtain the stats again and call addInfightOperation. That should not happen anymore. Move this into the mediator implementation and avoid unnecesary interactions with the stats object

Reply:
Moved it to the mediator, but the call to addInfightOperation is still necessary there to avoid breaking the api.